bin/smack.py

```diff
@@ -15,13 +15,71 @@
 import sys
 import tempfile
 from threading import Timer
+from toSVCOMPformat import *
+from token_replace import *
 
 VERSION = '1.5.1'
 temporary_files = []
 
+def rewriteIExtensionToC(args):
+  """ For svcomp mode, if file extension ends in .i, we need to make a copy as
+a .c file.  If we don't, clang treats the file as if preprocessing has already 
+occurred (and so doesn't run preprocessor)"""
+  #TODO check if there is a clang switch that causes clang to run preprocessor
+  #     even if file extension is .i
+  fileName = os.path.splitext(os.path.basename(args.input_file))[0]
+  newFileName = os.path.join(args.bcFolder, fileName) + '.original.c'
+  shutil.copyfile(args.input_file, newFileName)
+  return newFileName
+
+def replacer(args):
+  inputFileName = args.input_file
+  errorWitnessFileName = args.error_witness
+  outputFileName = args.bpl_file
+
+  fileName, fileExtension = os.path.splitext(os.path.basename(inputFileName))
+  with open(inputFileName, "r") as inputFile:
+    inputStr = inputFile.read()
+
+  """ If error witness flag is enabled, do tokenizing """
+  """ First get rid of these patterns which cause errors """
+  inputStr = re.sub(r'#line .*', '', inputStr)
+  inputStr = re.sub(r'# \d+.*', '', inputStr)
+  inputStr = re.sub(r'#pragma .*','',inputStr)
+  inputStr = beforeTokenReplace(inputStr)
+  """ Save valid tokens a tmp file in the folder containing bc files """
+  """ since tokenizer binary only accepts file as argument """
+  beforeTokenizedName = os.path.join(args.bcFolder, fileName) + '.tmp'
+  with open(beforeTokenizedName, 'w') as replacedFile:
+    replacedFile.write(inputStr)
+  tokenizedName = os.path.join(args.bcFolder, fileName) + '.tokenized.c'
+  cmd = ['tokenizer', beforeTokenizedName]
+
+  try:
+    with open(tokenizedName, 'w') as tokenizedFile:
+      proc = subprocess.Popen(cmd, preexec_fn=os.setsid, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+      output = proc.communicate()[0]
+      output = afterTokenReplace(output)
+      tokenizedFile.write(output)
+      rc = proc.returncode
+      proc = None
+    if rc:
+      raise RuntimeError("%s returned non-zero." % cmd[0])
+  except (RuntimeError, OSError) as err:
+    if output:
+      print >> sys.stderr, output
+    sys.exit("Error invoking command:\n%s\n%s" % (" ".join(cmd), err))
+  finally:
+    if proc: os.killpg(os.getpgid(proc.pid), signal.SIGKILL)
+
+  temporary_files.append(beforeTokenizedName)
+  args.input_file = tokenizedName
+  return args
+
 def frontends():
   """A dictionary of front-ends per file extension."""
   return {
+    '.i': clang_frontend,
```

Comment:
No.  *Most* of the svcomp benchmarks are .c, not .i.  The .i files are used when benchmarks require several includes for typedefs - rather than requiring competing tools to handle includes, the preprocessor alone is executed by the competition organizers, who then deliver a .i file to competitors, where the includes have already been included.